docs/settings.rst

```diff
@@ -0,0 +1,205 @@
+About settings
+==============
+
+.. tip::
+
+    We provide a way to autoconfigure many settings using your identity provider autoconfiguration endpoints.
+
+
+.. tip::
+    We provide ``Provider`` classes which implement setting generation for some popular identity providers (Keycloak, etc.).
+
+
+All those settings must be defined in ``settings.py`` under the variable name ``DJANGO_PYOIDC``.
+You should define them as a nested dictionary. The key to this dictionary is called your **provider name** (or **op_name** in some places). All your settings configuration are local to this provider. This allows multi-provider configurations.
+
+.. important::
+
+    Manually defined settings replaces settings from providers.
+
+Identity provider configuration
+===============================
+
+.. _provider-class-setting:
+
+provider_class
+**************
+
+**Default** : "django_pyoidc.providers.Provider"
+
+Use this setting to plug in a provider class that will be used to generate the settings of your identity provider.
+Some providers expect to receives custom arguments : you should defines them as settings.
+
+For example, the ``Keycloak10Provider`` can use two arguments : ``keycloak_base_uri`` and ``keycloak_realm``. If you wish to use them, define them in ``DJANGO_PYOIDC`` as your identity provider setting.
+
+provider_discovery_uri
+**********************
+
+This settings should be the URL of an OIDC autoconfiguration endpoint. We will use this
+setting to discover and store all the URLs needed to perform user authentication.
+
+client_secret
+*************
+
+This setting configures the client secret used to authenticate your application with an identity provider.
+
```

Comment:
Note: if you only have a client_id and not client_secret it means your OIDC client (application) was defined as a `public` application, which is normally only done for javascript SLA applications. A regular web application should have a client_secret, and a API backend application also.